main/src/ca/uwaterloo/flix/runtime/CompilationResult.scala

```diff
@@ -16,45 +16,41 @@
 
 package ca.uwaterloo.flix.runtime
 
-import ca.uwaterloo.flix.language.ast.ReducedAst.*
 import ca.uwaterloo.flix.language.ast.*
+import ca.uwaterloo.flix.language.ast.shared.Source
+import ca.uwaterloo.flix.language.phase.jvm.{JvmClass, JvmName}
 
 /**
   * A class representing the result of a compilation.
   *
-  * @param root      the abstract syntax tree of the program.
   * @param main      the reflected main function, if present.
-  * @param defs      the definitions in the program.
+  * @param tests     the tests in the program.
+  * @param sources   the sources of the program.
+  * @param classes   the JVM classes of the program.
   * @param totalTime the total compilation time, excluding class writing/loading.
-  * @param codeSize  the number of bytes the compiler generated.
   */
-class CompilationResult(root: Root,
-                        main: Option[Array[String] => Unit],
-                        defs: Map[Symbol.DefnSym, () => AnyRef],
-                        val totalTime: Long,
-                        val codeSize: Int) {
-
-  /**
-    * Optionally returns the main function.
-    */
-  def getMain: Option[Array[String] => Unit] = main
-
-  /**
-    * Returns all the test functions in the program.
-    */
-  def getTests: Map[Symbol.DefnSym, TestFn] = {
-    defs.collect {
-      case (sym, run) if root.defs(sym).ann.isTest =>
-        val skip = root.defs(sym).ann.isSkip
-        sym -> TestFn(sym, skip, run)
-    }
-  }
+class CompilationResult(main: Option[Array[String] => Unit],
```

Comment:
I would like this class restored to its old definition.

Reply:
Later those changes could come or not, but not in this PR

Reply:
I would like CompilationResult to be independent of the backend. I would also restore codesize. Why would it not be free to compute?

Reply:
> I would also restore codesize. Why would it not be free to compute?

I was trying to be consistent with getTotalLines and avoid collecting metrics eagerly. I think that is better. But i've reverted it.

> I would like CompilationResult to be independent of the backend

Is it good now?